test/compress/harmony.js

```diff
@@ -804,3 +804,21 @@ object_spread_of_sequence: {
         console.log({ ...o || o });
     }
 }
+
+// issue 2316
+class_name_can_be_preserved_with_reserved: {
+    mangle = {
+        reserved: ['Foo']
+    }
+    input: {
+        function x() {
+            class Foo {};
+            Foo.bar;
+            class Bar {};
+            Bar.foo;
+        }
```

Comment:
Could you please add a couple of class expressions to the test?
```
        function y() {
            var Foo = class Foo {};
            Foo.bar();
            var Bar = class Bar {};
            Bar.bar();
        }
```

Reply:
no worries, I do it... done ;)